src/barril/units/_tests/test_legacy_unit.py

```diff
@@ -0,0 +1,65 @@
+from barril.units import Scalar
+from pytest import approx, raises
+
+
+def testConvertLegacyUnit():
+    from barril.units.unit_database import LEGACY_TO_CURRENT
+
+    # test creating scalar using legacy representation and default category
+    q = Scalar(1.0, "1000ft3/d")
+    assert q.GetUnit() == "Mcf/d"
+
+    # test creating scalar using legacy representation
+    q = Scalar(category="volume flow rate", value=1.0, unit="1000ft3/d")
+    assert q.GetUnit() == "Mcf/d"
+
+    # test getting value using legacy representation
+    assert approx(q.GetValue("M(ft3)/d")) == q.GetValue("MMcf/d")
+
+    q = Scalar(category="volume flow rate", value=1.0, unit="M(ft3)/d")
+    assert q.GetUnit() == "MMcf/d"
+    assert approx(q.GetValue("M(ft3)/d")) == q.GetValue("MMcf/d")
+
+    # Test all possible legacy formats
+    for legacy, current in LEGACY_TO_CURRENT.items():
+        assert Scalar(1.0, legacy).GetUnit() == current
+
+
+def testCreateScalarUnitsError():
+    from barril.units.unit_database import UnitsError
+
+    with raises(UnitsError) as units_error:
+        _ = Scalar(1.0, "foo/d")
+    assert str(units_error.value) == "Unable to get default category for: foo/d"
+
+
+def testGetValueInvalidUnitError():
+    from barril.units.unit_database import InvalidUnitError
+
+    with raises(InvalidUnitError) as invalid_unit_error:
+        q = Scalar(1.0, "1000ft3/d")
+        q.GetValue("1000ft3/foo")
+    assert "Invalid unit for quantity_type volume flow rate: 1000ft3/foo" in str(
+        invalid_unit_error.value
+    )
```

Comment:
Can be written as:

```python
    with raises(InvalidUnitError, match="Invalid unit for quantity_type volume flow rate: 1000ft3/foo"):
        q = Scalar(1.0, "1000ft3/d")
        q.GetValue("1000ft3/foo")
```

Also, which one of those calls (Scalar creation/GetValue) you expect the error to come from? I suspect both, so you need two `pytest.raises`.